### PR TITLE
[wip] Update backward-compatible versions for checkpointing

### LINKED ISSUE #12475: Recent legacy checkpoints are not tested in CI
## :bug: Bug
Currently, only legacy checkpoints generated with versions from v1.0.0 to v1.4.1 are tested in CI, which are the versions listed in [legacy/back-compatible-versions.txt](https://github.com/PyTorchLightning/pytorch-lightning/blob/2e5728a4841cb1d3b75123ac941cd36f681f9a11/legacy/back-compatible-versions.txt).

For example, see this run on 1.6.0rc1: https://github.com/PyTorchLightning/pytorch-lightning/runs/5695770164?check_suite_focus=true#step:17:584

```
tests/checkpointing/test_legacy_checkpoints.py::test_load_legacy_checkpoints[1.0.0] PASSED [ 18%]
tests/checkpointing/test_legacy_checkpoints.py::test_load_legacy_checkpoints[1.0.1] PASSED [ 18%]
tests/checkpointing/test_legacy_checkpoints.py::test_load_legacy_checkpoints[1.0.2] PASSED [ 18%]
...
tests/checkpointing/test_legacy_checkpoints.py::test_load_legacy_checkpoints[1.3.8] PASSED [ 19%]
tests/checkpointing/test_legacy_checkpoints.py::test_load_legacy_checkpoints[1.4.0] PASSED [ 19%]
tests/checkpointing/test_legacy_checkpoints.py::test_load_legacy_checkpoints[1.4.1] PASSED [ 19%]
tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.0.0] PASSED [ 19%]
tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.0.1] PASSED [ 19%]
tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.0.2] PASSED [ 19%]
...
tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.3.8] PASSED [ 20%]
tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.4.0] PASSED [ 21%]
tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.4.1] PASSED [ 21%]
```

The file was introduced in #8535 and it hasn't been updated at all since then.

### Expected behavior
Run tests on legacy checkpoints generated with versions from v1.0.0 to the latest version in CI.

### Environment
n/a

### Additional context
Found this issue while reviewing/documenting our CI/CD.

cc @tchaton @rohitgr7 @akihironitta @carmocca @borda @awaelchli @ananthsub @ninginthecloud

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #12475.

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — COMMENTED** [review]
Have you created related version checkpoints?
btw, we have an action for you where you shall be just passing the particular version:
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/workflows/legacy-checkpoints.yml
see: https://github.com/PyTorchLightning/pytorch-lightning/actions/workflows/legacy-checkpoints.yml

**Comment by akihironitta** [comment]
@Borda No, I haven't. Let me try to run the action now! :]

File: legacy/generate_checkpoints.sh [hunk 1]
```diff
@@ -5,37 +5,38 @@
...
-  pip install "pytorch_lightning==$ver" "torch==$FROZEN_MIN_PT_VERSION" "torchmetrics" "scikit-learn" --quiet
+  pip install "pytorch_lightning==$pl_ver" "torch" "scikit-learn" --quiet
 
   python --version
   pip --version
   pip list | grep torch
 
-  python "$LEGACY_PATH/simple_classif_training.py"
-  cp "$LEGACY_PATH/simple_classif_training.py" ${LEGACY_PATH}/checkpoints/${ver}
+  python -X faulthandler "$LEGACY_PATH/simple_classif_training.py"
```
Comment (akihironitta) [inline legacy/generate_checkpoints.sh:35]: This is for segfault investigation: https://docs.python.org/3/library/faulthandler.html

**Comment by akihironitta** [comment]
Closing this in favour of https://github.com/PyTorchLightning/pytorch-lightning/pull/12491.

**Pull request closed without merging**